sql/log_event.cc

```diff
@@ -5216,6 +5216,13 @@ int Query_log_event::do_apply_event(Relay_log_info const *rli,
   thd->m_statement_psi= NULL;
   thd->m_digest= NULL;
 
+  /*
+    Prevent rewritten query from getting "stuck" in SHOW PROCESSLIST,
+    and performance_schema.threads.
+  */
+  thd->rewritten_query.mem_free();
+  thd->reset_query_for_display();
```

Comment:
On master, when create user is executed, how is freed? and is reset_query_for_display() is called?

Reply:
it's free unconditionally via `thd->rewritten_query.mem_free();` .
Regarding `thd->reset_query_for_display();` this is currently not been called, I've raised a bug which has been confirmed - https://bugs.mysql.com/bug.php?id=99204